lib/Cleantalk/Common/Transaction.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Cleantalk\Common;
+
+abstract class Transaction
+{
+    /**
+     * @var Transaction
+     */
+    private static $instance;
+
+    /**
+     * Time needed to perform an action
+     *
+     * @var int
+     */
+    private $action_time = 5;
+
+    /**
+     * Transaction ID option name
+     *
+     * @var string
+     */
+    private $tid_option_name;
+
+    /**
+     * @var string Option name with a start time of a transaction
+     */
+    private $start_time_option_name;
+
+    /**
+     * Constructor
+     *
+     * @return Transaction
+     */
+    public static function getInstance($params)
```

Comment:
Why do you duplicate the code from the Singleton trait instead of using it?

Reply:
Done https://github.com/CleanTalk/wordpress-antispam/pull/99/commits/6ce5f0a8f80746fa80a30327c41087a8022e1f7f